jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java

```diff
@@ -361,6 +376,97 @@ private synchronized float[][] predict(DMatrix data,
     return predicts;
   }
 
+  /**
+   * Perform thread-safe prediction.
+   *
+   * @param data      Flattened input matrix of features for prediction
+   * @param nrow      The number of preditions to make (count of input matrix rows)
+   * @param ncol      The number of features in the model (count of input matrix columns)
+   * @param missing   Value indicating missing element in the <code>data</code> input matrix
+   *
+   * @return predict  Result matrix
+   */
+  public float[][] inplace_predict(float[] data,
```

Comment:
@wbo4958 I just learned about the existence of `BigDenseMatrix`, do you think we should return the prediction in that?